llvm/lib/CodeGen/SelectionDAG/LegalizeIntegerTypes.cpp

```diff
@@ -875,6 +878,23 @@ SDValue DAGTypeLegalizer::PromoteIntRes_LOAD(LoadSDNode *N) {
   return Res;
 }
 
+SDValue DAGTypeLegalizer::PromoteIntRes_VP_LOAD(VPLoadSDNode *N) {
+  assert(!N->isIndexed() && "Indexed vp_load during type legalization!");
+  EVT NVT = TLI.getTypeToTransformTo(*DAG.getContext(), N->getValueType(0));
```

Comment:
if the NVT is same as N->getValueType(0) , do we still need to create a new SDValue with the NVT ?

Reply:
do we want to add an assert here ?

Reply:
I don't see much point. It's fundamental to how the type legalizer works and we don't assert it anywhere else.